flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/OneInputStreamTaskTest.java

```diff
@@ -597,6 +603,54 @@ public void testQuiesceTimerServiceAfterOpClose() throws Exception {
 		timeService.shutdownService();
 	}
 
+	@Test
+	public void testHandlingEndOfInput() throws Exception {
+		final OneInputStreamTaskTestHarness<String, String> testHarness = new OneInputStreamTaskTestHarness<>(
+			OneInputStreamTask::new,
+			2, 2,
+			BasicTypeInfo.STRING_TYPE_INFO, BasicTypeInfo.STRING_TYPE_INFO);
+
+		testHarness
+			.setupOperatorChain(new OperatorID(), new TestBoundedOneInputOperator("Operator0"))
+			.chain(
+				new OperatorID(),
+				new TestBoundedOneInputOperator("Operator1"),
+				BasicTypeInfo.STRING_TYPE_INFO.createSerializer(new ExecutionConfig()))
+			.finish();
+
+		ConcurrentLinkedQueue<Object> expectedOutput = new ConcurrentLinkedQueue<>();
+
+		testHarness.invoke();
+		testHarness.waitForTaskRunning();
+
+		TestBoundedOneInputOperator headOperator = (TestBoundedOneInputOperator) testHarness.getTask().headOperator;
+
+		testHarness.processElement(new StreamRecord<>("Hello-0-0"), 0, 0);
```

Comment:
Why are you passing `inputGate` and `channel` parameters? does it matter from where did the element come from?

Reply:
This test was written a long time ago, when they was needed. Now `inputGate` and `channel` parameters are not needed, and I will delete them.